### ISSUE TITLE
Does the `range` condition for integer(i64) is precise?

### ISSUE THREAD

**Issue by kobe2000** [qdrant/qdrant#5389]
Hi,

At first I looked for a method to filter string field in range, after reading the document, I realized there's only range condition for integer(i64) and float(f64).

So I plan to encode string into i64 to reach the goal, but I found the API only support double as parameter, I wonder whether it will be precisely correct when large integer is used

e.g.
```json
{
  "key": "price",
  "range": {
    "gte": 9223372036854775806,
    "lte": 9223372036854775807
  }
}
```

**Comment by kobe2000** [comment]
I find even the payload i64 is not precise!

I put a `346218760129241550`, but `346218760129241540` is saved

**Comment by timvisee** [comment]
I'm afraid this may be due to limitations of JavaScript/JSON.

JavaScript does have 53 bit numbers, not 64 bit numbers. It means that if using JavaScript, these numbers get cut off like you're seeing.

Are you indeed using JavaScript in the equation? If so, I'm afraid the recommendation will be to use smaller numbers or a different client.

**Comment by kobe2000** [comment]
> I'm afraid this may be due to limitations of JavaScript/JSON.
> 
> JavaScript does have 53 bit numbers, not 64 bit numbers. It means that if using JavaScript, these numbers get cut off like you're seeing.
> 
> Are you indeed using JavaScript in the equation? If so, I'm afraid the recommendation will be to use smaller numbers or a different client.

I use Java/grpc API, and I reproduce the problem by Qdrant web console too.

While I didn't see the source code, now I guess the i64 is backed by f64, so the integer can only use 52 bits (mantissa)

**Comment by timvisee** [comment]
> now I guess the i64 is backed by f64, so the integer can only use 52 bits (mantissa)

Correct. Note that this isn't Qdrant specific, and we don't really have control over this.

**Comment by kobe2000** [comment]
> > now I guess the i64 is backed by f64, so the integer can only use 52 bits (mantissa)
> 
> Correct. Note that this isn't Qdrant specific, and we don't really have control over this.

You mean that's designed so in Qdrant's implementation? Or just the API problem? I see only the double type in `io.qdrant.client.grpc.Points.Range` of Java/grpc client

**Comment by generall** [comment]
I don't think I can reproduce it 
![image](https://github.com/user-attachments/assets/cdd48136-5f2b-4f51-a516-68a67ce45e17)

**Comment by generall** [comment]
We fixed this problem in console by using different serializer, and it should not be the case for java/grpc client

**Comment by generall** [comment]
For range, though, it is a problem, cause range operated with f64 numbers, and it can't represent full spectrum of i64

**Comment by kobe2000** [comment]
> I don't think I can reproduce it ![image](https://private-user-images.githubusercontent.com/1935623/383913585-cdd48136-5f2b-4f51-a516-68a67ce45e17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwMjc0NzEsIm5iZiI6MTczMTAyNzE3MSwicGF0aCI6Ii8xOTM1NjIzLzM4MzkxMzU4NS1jZGQ0ODEzNi01ZjJiLTRmNTEtYTUxNi02OGE2N2NlNDVlMTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDhUMDA1MjUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjA5ZDE0MDk4ZDg1YzNhZGJhODBmNTc3ZGZmYmQ4MDk1ZWY1N2Y0MDIwNTNhODk3ODNmNDM3M2M4NmQ2NTg4NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.HGVK1wXWuDcOZ-NtFkZ-fmrrH_-ym8t5LSZ2VQMeWK4)

I was wrong in my expression of `console`, in deed I was using the Edit Payload funciton in page `/dashboard#/collections/{name}`

**Comment by kobe2000** [comment]
> For range, though, it is a problem, cause range operated with f64 numbers, and it can't represent full spectrum of i64

Then would you plan to fix it?

**Comment by generall** [comment]
> > For range, though, it is a problem, cause range operated with f64 numbers, and it can't represent full spectrum of i64
> 
> Then would you plan to fix it?

The fix is not trivial, as grpc doesn't support larger floats.
